ui/js/services/main.js

```diff
@@ -128,17 +128,35 @@ treeherder.factory('thNotify', [
              * @sticky is a boolean indicating if you want the message to disappear
              * after a while or not
              */
-            send: function (message, severity, sticky, linkText, url) {
+            send: function (message, severity, opts) {
                 $log.debug("received message", message);
+                opts = opts || {};
                 severity = severity || 'info';
-                sticky = sticky || false;
+                var sticky = opts.sticky || false;
+
+                if (!_.isPlainObject(opts)) {
+                    // In this case a developer has passed in an old-style
+                    // sticky arg. We will notify them during development that
+                    // there is a new way of doing this but still display the
+                    // original notification they wanted to display.
+                    thNotify.notifications.unshift({
+                        message: 'You are using an old notification style!',
+                        severity: 'danger',
+                        sticky: true,
+                        linkText: 'Check bug 1402062 for the new style',
+                        url: 'https://bugzilla.mozilla.org/show_bug.cgi?id=1402062',
+                        created: Date.now(),
+                    });
+                    sticky = opts;
+                }
+
                 var maxNsNotifications = 5;
                 var notification = {
                     message: message,
                     severity: severity,
                     sticky: sticky,
-                    linkText: linkText,
-                    url: url,
+                    linkText: opts.linkText,
```

Comment:
I know you were trying to change less code (which I appreciate.  :) ).  But I wonder if it'd be nicer to just clean up this bit with a tiny refactor in ES6:

    thNotify.notifications.unshift({ ...opts, message, severity, created: Date.now() };